src/bio/bio_context.c

```diff
@@ -176,12 +268,12 @@ bio_blob_create(uuid_t uuid, struct bio_xs_context *xs_ctxt, uint64_t blob_sz)
 
 	D_ASSERT(xs_ctxt != NULL);
 	bbs = xs_ctxt->bxc_blobstore;
-	D_ASSERT(bbs != NULL);
+	rc = bio_check_bs(bbs);
+	if (rc)
+		return rc;
 
-	ABT_mutex_lock(bbs->bb_mutex);
```

Comment:
was this lock unnecessary before, or did something change here?

Reply:
It was necessary before. This patch changed the blobstore closing code to guarantee that all blobstore users put blobstore first, then blobstore owner close blobstore, it's not necessary anymore.